src/app/shared/refresh.service.ts

```diff
@@ -90,6 +90,28 @@ export class RefreshService {
     );
   }
 
+  /**
+   * Refresh an individual version of a workflow and optionally updates the GA4GH files
+   * @param workflowId
+   * @param versionName
+   * @memberof RefreshService
+   */
+  refreshWorkflowVersion(workflowId: string, versionName: string): void {
+    const message = 'Refreshing ' + this.workflow.full_workflow_path + ' version ' + versionName;
```

Comment:
It was a mistake for this service to subscribe to another service. Avoid using `this.workflow` so that it's easier to remove, just pass it in from the component or workflowQuery.getValue().

Reply:
Do you think it is worth creating an issue to fix this service subscribing to a service?

Reply:
wouldn't hurt I guess